python/langsmith/_internal/_background_thread.py

```diff
@@ -88,6 +96,42 @@ def _tracing_thread_drain_queue(
     return next_batch
 
 
+def _tracing_thread_drain_compressed_buffer(
+    client: Client, size_limit: int = 100, size_limit_bytes: int | None = 20_971_520
+) -> Optional[io.BytesIO]:
+    assert client.compressed_runs_buffer is not None
+    assert client.compressor_writer is not None
+    with client._buffer_lock:
+        current_size = client.compressed_runs_buffer.tell()
+
+        if size_limit is not None and size_limit <= 0:
+            raise ValueError(f"size_limit must be positive; got {size_limit}")
+        if size_limit_bytes is not None and size_limit_bytes < 0:
+            raise ValueError(
+                f"size_limit_bytes must be nonnegative; got {size_limit_bytes}"
+            )
+
+        if (size_limit_bytes is None or current_size < size_limit_bytes) and (
+            size_limit is None or client._run_count < size_limit
```

Comment:
Why 100?

Reply:
This is the limit that is currently used. It also performed well in flush time benchmarks.

Reply:
It can also be overridden by the batch ingest config